openedx/core/djangoapps/content/learning_sequences/data.py

```diff
@@ -116,17 +119,34 @@ def user_partition_groups_not_empty(instance, attribute, value):  # pylint: disa
 
 
 @attr.s(frozen=True)
-class CourseLearningSequenceData:
+class LearningSequenceData:
     """
-    A Learning Sequence (a.k.a. subsection) from a Course.
+    A generic, context-agnostic Learning Sequence.
 
-    It's possible that at some point we'll want a LearningSequenceData
-    superclass to encapsulate the minimum set of data that is shared between
-    learning sequences in Courses vs. Pathways vs. Libraries. Such an object
-    would likely not have `visibility` as that holds course-specific concepts.
+    This class does not contain any context-specific (that is, Course-specific,
+    Pathway-specific, etc) data.
     """
+    class DoesNotExist(ObjectDoesNotExist):
+        pass
+
     usage_key = attr.ib(type=UsageKey)
     title = attr.ib(type=str)
+
+    # A URL-safe Base64-encoding of a blake2b hash of the usage key.
+    # For aesthetic use (eg, as a path parameter, to shorten URLs).
+    # This field is experimental. See TNL-8638 for more information.
+    usage_key_hash = attr.ib(type=str)
```

Comment:
Nit: Might want to specify in the comment whether padding `=` characters are stripped or not.

Reply:
Will do.